widgets/src/widgets/base.ts

```diff
@@ -0,0 +1,29 @@
+import type { ConnectionHandler } from '@/network'
+import type { CartWidgetParams } from '@/interfaces'
+
+export interface WidgetConfig {
+  containerSelector: string
+  api: ConnectionHandler
+}
+
+export class Widget {
+  api: ConnectionHandler
+  containerSelector: string
```

Comment:
It is optional for the sdk consumer, because it's optional to `class GreensparkWidgets`, but it is not optional on the context  of the `Widget`. This way the default value is established in the parent of each Widget, and not in the Widget itself.

Reply:
gotcha